modules/stack/dependent_stacks.tf

```diff
@@ -0,0 +1,16 @@
+
+locals {
+  map_of_labels_array = {
+    for label in var.labels : split(":", label)[0] => split(":", label)[1]... # the ellipsis creates a group of values
+  }
+  depends_on_labels = local.map_of_labels_array["depends_on"]
+  non_depends_on_labels = [
+    for label in var.labels : label if !startswith(label, "depends_on:")
+  ]
+}
+
+resource "spacelift_stack_dependency" "dependent_stacks" {
+  for_each            = var.enabled && var.use_depends_on_resource ? toset(local.depends_on_labels) : []
+  stack_id            = spacelift_stack.default[0].id
+  depends_on_stack_id = each.value
```

Comment:
So it looks like the spacelift provider will do this hookup if you give it the slug. I'm resolving this convo, since maybe Jake was just not aware of the ability of the provider to handle it.